synapse/config/jwt.py

```diff
@@ -43,11 +43,11 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             self.jwt_audiences = jwt_config.get("audiences")
 
             try:
-                import jwt
+                from authlib.jose import JsonWebToken
 
-                jwt  # To stop unused lint.
+                JsonWebToken  # To stop unused lint.
             except ImportError:
-                raise ConfigError(MISSING_JWT)
+                raise ConfigError(MISSING_AUTHLIB)
```

Comment:
I'd prefer this to be a  call to `check_dependencies` (which checks the version of authlib installed). Additionally this shouldn't raise a ConfigError: the config may be perfectly valid; it's your installation that's screwed up.

However there's already a lot going on here, and there are other placed with the same pattern. I'll raise a new issue for this.

Reply:
That's fair. #13066, for completeness